catalog/core/catalog-core-solrcommands/src/test/java/org/codice/ddf/commands/solr/RestoreCommandTest.java

```diff
@@ -0,0 +1,714 @@
+    assertThat(consoleOutput.getOutput(), containsString("Restore complete."));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testPerformSolrCloudSynchronousRestoreNoOptions() throws Exception {
+    setupSystemProperties(SolrCommands.CLOUD_SOLR_CLIENT_TYPE);
+    RestoreCommand restoreCommand =
+        getSynchronousRestoreCommand(null, null, miniSolrCloud.getSolrClient());
+    restoreCommand.execute();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testPerformSolrCloudSynchronousRestoreNoBackupLocation() throws Exception {
+    setupSystemProperties(SolrCommands.CLOUD_SOLR_CLIENT_TYPE);
+    RestoreCommand restoreCommand =
+        getSynchronousRestoreCommand(null, DEFAULT_CORE_NAME, miniSolrCloud.getSolrClient());
+    restoreCommand.execute();
+  }
+
+  @Test
+  public void testPerformSolrCloudSynchronousRestoreNoCollection() throws Exception {
+    setupSystemProperties(SolrCommands.CLOUD_SOLR_CLIENT_TYPE);
+    RestoreCommand restoreCommand =
+        getSynchronousRestoreCommand(getBackupLocation(), null, miniSolrCloud.getSolrClient());
+    restoreCommand.execute();
+
+    String backupName = getBackupName(consoleOutput.getOutput());
+    File backupFile =
+        Paths.get(restoreCommand.backupLocation, backupName).toAbsolutePath().toFile();
+    assertThat(
+        consoleOutput.getOutput(),
+        containsString(
+            String.format(
+                "Restoring collection [%s] from [%s] / [%s",
+                DEFAULT_CORE_NAME, restoreCommand.backupLocation, backupName)));
+    assertThat(consoleOutput.getOutput(), containsString("Restore complete."));
+    assertThat(backupFile.exists(), is(true));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testPerformSolrCloudAsynchronousRestoreWithAsyncStatusOptionsSupplied()
+      throws Exception {
+    setupSystemProperties(SolrCommands.CLOUD_SOLR_CLIENT_TYPE);
+    RestoreCommand restoreCommand =
+        getRestoreCommand(
+            getBackupLocation(),
+            DEFAULT_CORE_NAME,
+            true,
+            true,
+            "myRequestId1",
+            miniSolrCloud.getSolrClient());
+    restoreCommand.execute();
+  }
+
+  @Test
+  public void testPerformSolrCloudAsynchronousRestore() throws Exception {
+    setupSystemProperties(SolrCommands.CLOUD_SOLR_CLIENT_TYPE);
+    RestoreCommand restoreCommand =
+        getAsnychronousRestoreCommand(
+            getBackupLocation(), DEFAULT_CORE_NAME, miniSolrCloud.getSolrClient());
+    restoreCommand.execute();
+
+    assertThat(
+        consoleOutput.getOutput(),
+        containsString(
+            String.format(
+                "Restoring collection [%s] from [%s] / [%s_",
+                DEFAULT_CORE_NAME, restoreCommand.backupLocation, DEFAULT_CORE_NAME)));
+    assertThat(consoleOutput.getOutput(), containsString("Restore request Id:"));
```

Comment:
Is there any way we can set up the test to return a specific request ID and check for it here?

Reply:
Not really, as the request Id comes from Solr when we tell it to do the restore asynchronously.